opensearch/src/main/java/org/opensearch/sql/opensearch/storage/script/filter/lucene/relevance/MatchQuery.java

```diff
@@ -63,6 +63,9 @@ public class MatchQuery extends LuceneQuery {
 
   @Override
   public QueryBuilder build(FunctionExpression func) {
+    if (func.getArguments().size() < 2) {
```

Comment:
This would be expanding the scope of this PR.


I'm not sure if `canSupport` as written is appropriate here. We'd have to  override it in each subclass of LuceneQuery and move the relevant piece of logic there.

Reply:
It doesn't need to be duplicated to subclasses, as subclasses will inherit the function. 
Whether it's appropriate... calls to build() seem to already check that the canSupport conditions are met. So it's a condition to cover new cases.  This last point probably needs to be verified.

Reply:
I'd leave it till later -- there are other issues with that file.

In particular, `canSupport` is already called in `org/opensearch/sql/opensearch/storage/script/filter/FilterQueryBuilder.java:83`
... which then makes this size check kind of redundant.

Overriding `canSupport` and specializing it for each subclass of LuceneQuery would make sense but kind of creeping the scope here.

Reply:
Ok - but it makes me a little concerned that we might be missing a case, or duplicating business logic.